server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -0,0 +1,771 @@
+            return getTextSearchInfo().getSearchAnalyzer().normalize(name(), value.toString());
+        }
+
+        /**
+         * redirect queries with rewrite value to rewriteSearchValue and directSubFieldName
+         */
+        @Override
+        public Query termQuery(Object value, @Nullable QueryShardContext context) {
+
+            String searchValueString = inputToString(value);
+            String directSubFieldName = directSubfield();
+            String rewriteSearchValue = rewriteValue(searchValueString);
+
+            failIfNotIndexed();
+            Query query;
+            query = new TermQuery(new Term(directSubFieldName, indexedValueForSearch(rewriteSearchValue)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            return query;
+        }
+
+        @Override
+        public Query termsQuery(List<?> values, QueryShardContext context) {
+            failIfNotIndexed();
+            String directedSearchFieldName = directSubfield();
+            BytesRef[] bytesRefs = new BytesRef[values.size()];
+            for (int i = 0; i < bytesRefs.length; i++) {
+                String rewriteValues = rewriteValue(inputToString(values.get(i)));
+
+                bytesRefs[i] = indexedValueForSearch(new BytesRef(rewriteValues));
+
+            }
+
+            return new TermInSetQuery(directedSearchFieldName, bytesRefs);
+        }
+
+        /**
+         * To direch search fields, if a dot path was used in search query,
+         * then direct to flatObjectFieldName._valueAndPath subfield,
+         * else, direct to flatObjectFieldName._value subfield.
+         * @return directedSubFieldName
+         */
+        public String directSubfield() {
+            if (name().contains(DOT_SYMBOL)) {
+                String[] dotPathList = name().split("\\.");
+                return dotPathList[0] + VALUE_AND_PATH_SUFFIX;
+            } else {
+                return this.valueFieldType.name();
+            }
+        }
+
+        /**
+         * If the search key is assigned with value,
+         * the dot path was used in search query, then
+         * rewrite the searchValueString as the format "dotpath=value",
+         * @return rewriteSearchValue
+         */
+        public String rewriteValue(String searchValueString) {
+            if (!name().contains(DOT_SYMBOL)) {
+                return searchValueString;
+            } else {
+                String rewriteSearchValue = new StringBuilder().append(name()).append(EQUAL_SYMBOL).append(searchValueString).toString();
+                return rewriteSearchValue;
+            }
+
+        }
+
+        private String inputToString(Object inputValue) {
+            if (inputValue instanceof Integer) {
+                String inputToString = Integer.toString((Integer) inputValue);
+                return inputToString;
+            } else if (inputValue instanceof Integer) {
```

Comment:
Same check again? Probably not needed

Reply:
nice catch. removed the duplicated lines in [Add IT tests for painless query in testDocValues](https://github.com/opensearch-project/OpenSearch/pull/6507/commits/9fff5fda50199eb4b80d0df5debd38ea0047713b)